bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/models/v1/common/ProductListItemImpl.java

```diff
@@ -70,6 +75,29 @@ public ProductListItemImpl(String sku, String slug, String name, Price price, St
             : CommerceIdentifierImpl.fromProductSku(sku);
     }
 
+    public ProductListItemImpl(ProductInterface product, Page productPage, String activeVariantSku, SlingHttpServletRequest request,
```

Comment:
This is the third constructor for this class. I think we should consider using another pattern for constructing these objects, maybe a "Builder"?